pkg/cephfs/volumemounter.go

```diff
@@ -74,6 +74,31 @@ type volumeMounter interface {
 	name() string
 }
 
+func getDefaultMounter() string {
+	mounter := volumeMounterFuse
```

Comment:
So what if FuseMounter wasn't available? Instead may be you can check kernel version in loadAvailableMounters, and make add kernel mounter as available only when kernel version >=4.17?

Reply:
The newMounter takes care of checking, if the defauls/the requested mounter is available, if we check in load available, it will be duplicate check in newMounter too..

Reply:
Done

Reply:
A quick look. Wondering why `mounter` is needed here, if it is not changing.

Reply:
> The newMounter takes care of checking, if the defauls/the requested mounter is available, if we check in load available, it will be duplicate check in newMounter too..

So you're OK with kernel mounter being in the slice `availableMounters` even though the kernel version is less than 4.17?

I was thinking that you can get rid of this function `getDefaultMounter()`.

In `loadAvailableMounter()`, if kernel version >=4.17, let the kernel mounter be the first element in the splice  availableMounters`, else don't have kernel mounter in the splice `availableMounters` and log a warning that kernel is not an availableMounter since version kernel version < 4.17.

In `driver.go`'s `Run()`, you can leave `DefaultVolumeMounter = availableMounter[0]` as was before.

And when storage admin chooses kernel mounter via the storage class option and the setup has a kernel less than 4.17,  raise an error that the kernel version >= 4.17 is required for Ceph CSI.